pkg/daemon/update.go

```diff
@@ -50,6 +50,38 @@ const (
 	realtimeKernelType = "realtime"
 )
 
+// HostInfo contains information of an OSTree based system
+type HostInfo struct {
+	Deployments []Deployment `json:"deployments"`
+}
+
+// Deployment contains information about a particular OSTree deployment
+type Deployment struct {
+	RequestedLocalPkgs []string `json:"requested-local-packages"`
+}
+
+func installedRTKernelRpmsOnHost() ([]string, error) {
+	var out []byte
+	var err error
+	var rtKernelRpms = []string{}
+	if out, err = exec.Command("rpm-ostree", "status", "--json").Output(); err != nil {
+		return rtKernelRpms, fmt.Errorf("Failed to execute rpm-ostre status --json %v", err)
+	}
+
+	var rpms HostInfo
+	if err := json.Unmarshal(out, &rpms); err != nil {
+		return rtKernelRpms, err
+	}
+
+	rtRegex := regexp.MustCompile("kernel-rt-.*")
+	for _, localPkg := range rpms.Deployments[0].RequestedLocalPkgs {
+		if rtRegex.MatchString(localPkg) {
```

Comment:
This could just be `strings.HasPrefix(localPkg, "kernel-rt-")`.  But fine as is too.